src/MongoBson.pas

```diff
@@ -1473,7 +1473,7 @@ function TBsonBuffer.startObject(const Name: UTF8String): Boolean;
 var
   child: Pointer;
 begin
-  child := bson_new();
+  child := bson_malloc(SizeOf(bson_t));
```

Comment:
@alexeyvo , this is similar to what Eugene did. I don't have a problem doing it like this, what I don't like is the bson_t delphi record mapping the C struct.
If we need to do this, can you expose a function on the C driver just to return the sizeof(bson_t) ? In general I don't like the approach of dll consumer language declared structs